Prepare for 4.23.0

lukeheath opened the pull request.

noahtalerman commented:
Looks good!

@lukeheath FYI Chris and I are editing the CHANGELOG in the draft release article here ("More new features..." section): https://docs.google.com/document/d/1kD3sizH5xOZrPoGHB4393fcfe-g4rq9fg4Z1J5Y_QzY/edit#heading=h.qzzxovn8ejvw

I will update the CHANGELOG here in this PR when we wrap up these^ edits.

chris-mcgillicuddy reviewed (commented):
@noahtalerman, do you think there will be additional edits to the CHANGELOG?

noahtalerman commented:
> do you think there will be additional edits to the CHANGELOG?

@chris-mcgillicuddy it's likely that there will be additional edits to the CHANGELOG. 

This is because there have been several bugs caught during testing that may make it into the changelog. Check out the #help-release-bugs Slack channel (internal) here: https://fleetdm.slack.com/archives/C0459H8RN3C/p1668192641781889

I think we'll want these bugs to make it into the release article. 

@lukeheath can you please @ mention Chris if there are any changes to the CHANGELOG?

lukeheath commented:
@chris-mcgillicuddy There have been four additions to the changelog: 

* Fixed software filter when software does not have a version number.

* Fixed validation of agent options where valid options were being rejected (such as `exclude_paths`).

* Fixed styles for code editor gutter and query side panel table dropdown.

* Styling improvements to App Settings page.

chris-mcgillicuddy commented:
Thanks, @lukeheath!
